src/screens/Addresses/Contact/ContactFormBaseScreen.tsx

```diff
@@ -18,33 +18,41 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 
 import { Controller, useForm } from 'react-hook-form'
 
-import Button from '~/components/buttons/Button'
+import { ContinueButton } from '~/components/buttons/Button'
 import Input from '~/components/inputs/Input'
 import { ScreenSection } from '~/components/layout/Screen'
-import { AddressHash } from '~/types/addresses'
+import ScrollScreen, { ScrollScreenProps } from '~/components/layout/ScrollScreen'
 import { ContactFormData } from '~/types/contacts'
 import { isContactAddressValid, isContactNameValid } from '~/utils/form-validation'
 import { validateIsAddressValid } from '~/utils/forms'
 
-interface ContactFormProps {
+interface ContactFormProps extends ScrollScreenProps {
   initialValues: ContactFormData
   onSubmit: (data: ContactFormData) => void
   buttonText?: string
-  disableIsMainToggle?: boolean
-  addressHash?: AddressHash
 }
 
 const requiredErrorMessage = 'This field is required'
 
-const ContactForm = ({ initialValues, onSubmit, buttonText = 'Save' }: ContactFormProps) => {
+const ContactForm = ({ initialValues, onSubmit, buttonText = 'Save', headerOptions, ...props }: ContactFormProps) => {
   const {
     control,
     handleSubmit,
     formState: { errors }
   } = useForm<ContactFormData>({ defaultValues: initialValues })
 
   return (
-    <>
+    <ScrollScreen
+      usesKeyboard
+      fill
+      hasNavigationHeader
+      headerOptions={{
+        type: 'stack',
+        headerRight: () => <ContinueButton title={buttonText} onPress={handleSubmit(onSubmit)} />,
```

Comment:
We should use another icon than the right arrow for this one, as there's no step after saving. A checkmark could be better IMO. What do you think?

Reply:
Also true for the address edit screen and the new address screen

Reply:
Sure! I'm on it!